src/ConcreteOperations/isdisjoint.jl

```diff
@@ -446,33 +445,37 @@ center, and ``g_i`` are the zonotope's generators.
 For witness production we fall back to a less efficient implementation for
 general sets as the first argument.
 """
-function is_intersection_empty(Z::Zonotope{N},
-                               H::Hyperplane{N},
-                               witness::Bool=false
-                              ) where {N<:Real}
+function is_intersection_empty(Z::Zonotope{N}, H::Hyperplane{N}, witness::Bool=false) where {N<:Real}
     if witness
-        # use less efficient implementation that supports witness production
-        return invoke(is_intersection_empty,
-                      Tuple{LazySet{N}, Hyperplane{N}, Bool},
-                      Z, H, witness)
-    end
-
-    v = H.b - dot(H.a, Z.center)
-    p = ngens(Z)
-    if p == 0
-        abs_sum = zero(N)
+        return _is_intersection_empty(Z, H, Val(true))
     else
-        abs_sum = sum(abs(dot(H.a, Z.generators[:, i])) for i = 1:ngens(Z))
+        return _is_intersection_empty(Z, H, Val(false))
     end
-    return v < -abs_sum || v > abs_sum
 end
 
 # symmetric method
-function is_intersection_empty(H::Hyperplane{N},
-                               Z::Zonotope{N},
-                               witness::Bool=false
-                              ) where {N<:Real}
-    return is_intersection_empty(Z, H, witness)
+is_intersection_empty(H::Hyperplane{N}, Z::Zonotope{N}, witness::Bool=false) where {N<:Real} = is_intersection_empty(Z, H, witness)
+
+function _is_intersection_empty(Z::Zonotope{N}, H::Hyperplane{N}, ::Val{false}) where {N}
+    c, G = Z.center, Z.generators
+    v = H.b - dot(H.a, c)
+
+    n, p = size(G)
+    abs_sum = zero(N)
+    if p > 0
+        @inbounds for i in 1:n
+            aux = zero(N)
+            @simd for j in 1:p
+                aux += H.a[i] * G[j, i]
+            end
+            abs_sum += abs(aux)
```

Comment:
You can pull out the multiplication with `H.a[i]` which is a constant here.
And I think it should be `G[i, j]`, right?
```suggestion
            aux = zero(N)
            @simd for j in 1:p
                aux += G[i, j]
            end
            abs_sum += abs(H.a[i] * aux)
```

Reply:
good catch, that should have been `H.a[j] * G[j, i]`. the multiplication is from the left because we want to sum `|<a, g>|` for each generator `g`.

Reply:
I think this is wrong. `j` runs through the number of columns but you use it for the rows, and vice versa for `i`.

Reply:
i've changed the variable's names to match what we usually use.